### ISSUE TITLE
Feature request: note sharing with nextcloud users

### ISSUE THREAD

**Issue by menelic** [PhieF/CarnetDocumentation#30]
To make this a keep replacement, please enable note sharing, at least with users on the same nextcloud instance.evetyday example; create a shopping list as to do list and share it with family members

**Comment by PhieF** [comment]
This feature will require a lot of work, I'm afraid I'm not sure I will have time to implement it

**Comment by chepec** [comment]
If I might inquire, lots of work due to the coding effort involved, or due to overcoming some inherent limitations in the project's stack? I'm not a programmer myself, but I've been testing your project for a while (as a user and Nextcloud admin) and I find it very useful---a bit rough around the edges---but very promising.

So if it's just a matter of time and number of eyeballs on Carnet (or fingers on keyboards, as it were), then great! Considering that your project is the singular Keep-clone notes app for Nextcloud, I have high hopes for its popularity and will certainly help by spreading the word. But if there's some inherent technical limitation making Nextcloud sharing hard to implement (even with more time), I would really appreciate a heads up from you.

**Comment by menelic** [comment]
yes, please share the hurdles, I agree with chepec and would hope that nextcloud programmers might be wiling to support, given the fact that this is the only keep replacement and sharing is the only keep core feature lacking in carnet. As enabling sharing for nextclud apps is a key infrastructure issue with relevance for many other nextcloud apps I am optimistic that you could get assistance

**Comment by PhieF** [comment]
Hi, I will see what I can do but I fear huge conflicts on note edition, for example when both editing notes in the same time

**Comment by ovz93br43v7** [comment]
This is the only feature I need to switch from Google Notes. If it's technically stable, I would have no problem to buy it as a pro feature.

**Comment by PhieF** [comment]
I really do understand the need of this feature, but as I said it will take some time because I will soon have to get a full time paid job. Perhaps a share as read only feature would be easier at first

**Comment by downeym** [comment]
Would love to see this. I'm currently using the standard NC notes app, with a shared directory to contain the source .md files. Obviously the UI/UX of Carnet is much better, but need sharing in order to switch. 😄

**Comment by PhieF** [comment]
Well perhaps you can try to share the folder Documents/QuickDoc (or QuickNote ?) to try if it's enough

**Comment by downeym** [comment]
Cool, that seems to be the equivalent. Will try it out. Thanks.

**Comment by PhieF** [comment]
make a backup of your folder before trying that, I've never tested it

**Comment by jakob42** [comment]
I don't want to pressure you in any way, this is FOSS software after all, so do it in your pace and whatever features are useful to you. Anything else is a bonus.

Just for completeness, sharing locally would be really good and would give me the chance to migrate, since I only share in my family ATM. But it would be outstanding, if federated sharing would work as well. I understand that thats a lot more difficult (if only just because you cannot make sure that everybody uses the same version), but I wanted to put it out there.

**Comment by PhieF** [comment]
No problem, I do understand this is an important feature and I hope to be able to implement it correctly, I just have first to assess the difficulty of it before making any promises
